ws-server/src/server.rs

```diff
@@ -272,22 +278,38 @@ async fn background_task(
 	while !stop_server.shutdown_requested() {
 		data.clear();
 
-		if let Err(e) = method_executors.select_with(receiver.receive_data(&mut data)).await {
-			tracing::error!("Could not receive WS data: {:?}; closing connection", e);
-			tx.close_channel();
-			return Err(e.into());
-		}
+		if let Err(err) = method_executors.select_with(receiver.receive_data(&mut data)).await {
+			match err {
+				SokettoError::Closed => {
+					tracing::debug!("Remote peer terminated the connection: {}", conn_id);
+					tx.close_channel();
+					return Ok(());
+				}
+				SokettoError::MessageTooLarge { current, maximum } => {
+					tracing::warn!(
+						"WS transport error: message is too big error ({} bytes, max is {})",
+						current,
+						maximum
+					);
+					send_error(Id::Null, &tx, ErrorCode::OversizedRequest.into());
+					continue;
+				}
+				// These errors can not be gracefully handled, so just log them and terminate the connection.
+				err => {
+					tracing::error!("WS transport error: {:?} => terminate connection {}", err, conn_id);
+					tx.close_channel();
+					return Err(err.into());
+				}
+			};
+		};
 
-		if data.len() > max_request_body_size as usize {
-			tracing::warn!("Request is too big ({} bytes, max is {})", data.len(), max_request_body_size);
-			send_error(Id::Null, &tx, ErrorCode::OversizedRequest.into());
-			continue;
-		}
+		tracing::debug!("recv {} bytes", data.len());
 
 		match data.get(0) {
 			Some(b'{') => {
 				if let Ok(req) = serde_json::from_slice::<Request>(&data) {
-					tracing::debug!("recv: {:?}", req);
+					tracing::debug!("recv call={}", req.method);
+					tracing::trace!("recv: {:?}", req);
```

Comment:
```suggestion
					tracing::trace!("recv: req={:?}", req);
```

Reply:
the debug impls prints `Request { }` but fine :)